packages/remix-dev/vite/plugin.ts

```diff
@@ -1629,24 +1655,40 @@ async function getRouteMetadata(
   return info;
 }
 
-async function handleSpaMode(
-  serverBuildDirectoryPath: string,
-  serverBuildFile: string,
-  clientBuildDirectory: string,
+async function getPrerenderBuildAndHandler(
   viteConfig: Vite.ResolvedConfig,
-  basename: string
+  reactRouterConfig: Awaited<ReturnType<typeof resolveReactRouterConfig>>,
+  serverBuildDirectory: string
 ) {
-  // Create a handler and call it for the `/` path - rendering down to the
-  // proper HydrateFallback ... or not!  Maybe they have a static landing page
-  // generated from routes/_index.tsx.
-  let serverBuildPath = path.join(serverBuildDirectoryPath, serverBuildFile);
+  let serverBuildPath = path.join(
+    serverBuildDirectory,
+    reactRouterConfig.serverBuildFile
+  );
   let build = await import(url.pathToFileURL(serverBuildPath).toString());
   let { createRequestHandler: createHandler } = await import(
     "@react-router/node"
   );
-  let handler = createHandler(build, viteConfig.mode);
-  let response = await handler(new Request(`http://localhost${basename}`));
+  return {
+    build,
+    handler: createHandler(build, viteConfig.mode),
+  };
+}
+
+async function handleSpaMode(
+  viteConfig: Vite.ResolvedConfig,
+  reactRouterConfig: Awaited<ReturnType<typeof resolveReactRouterConfig>>,
+  serverBuildDirectory: string,
+  clientBuildDirectory: string
+) {
+  let { handler } = await getPrerenderBuildAndHandler(
+    viteConfig,
+    reactRouterConfig,
+    serverBuildDirectory
+  );
```

Comment:
SPA Mode is effectively untouched, just sharing some code for creating the handler with the prerender code paths